scripts/header_parser.py

```diff
@@ -452,19 +574,21 @@ def create_wrappers_for_header_file(path: str,
     c_wrappers = []
     ret = set()
     for func in extract_functions_from_file(path):
-        absolute_path = Path(path).absolute()
-        name, params = parse_function_info(func, absolute_path)
-        if namespace is None or name in namespace:
-            try:
+        func['rtnType'] = re.sub(type_qualifiers, '', func['rtnType'])
+        # we're only interested in scalar values
+        try:
+            absolute_path = Path(path).absolute()
+            name, params = parse_function_info(func, absolute_path)
+            if namespace is None or name in namespace:
```

Comment:
# in case NAMESPACE file is not provided (here `not namespace`) we generate wrappers for all functions
# otherwise wrappers are generated only for functions listed in NAMESPACE file (here `name in namespace`)
```suggestion
            if not namespace or name in namespace:
```

Reply:
An empty namespace (e.g. if we create wrappers but don't want to export any of them) is also falsly, hence `namespace is None`, I will add the comment though